src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplateSupport.java

```diff
@@ -218,4 +250,23 @@ private boolean canPublishEvent() {
 		return object;
 	}
 
+	public <T> Integer getTxResultKey(T source) {
+		final CouchbasePersistentEntity<?> persistentEntity = mappingContext.getRequiredPersistentEntity(source.getClass());
+		final CouchbasePersistentProperty transactionResultProperty = persistentEntity.transactionResultProperty();
+		if (transactionResultProperty == null) {
+			throw new TransactionUsageException(
+					"the TransactionGetResult was not found using the identityHashCode of the object, " + " and the class "
+							+ source.getClass().getSimpleName() + " does not have an @" + TransactionResultKey.class.getName()
+							+ " Integer field\n object: " + source);
```

Comment:
should we make that object an `Object` and document that the content is opaque? Or make it some kind of wrapper class we own (i.e. a TransactionMetaData) .. so that we can always change what's inside in the future?